core/src/logger.rs

```diff
@@ -1,11 +1,8 @@
-//use std::sync::{Arc, Mutex};
-
-/*!
-This logger is the logger that's attached to each Holochain application
-which is separate from standard logging via the log crate warn! info! debug! logging that
-gets emitted globaly from the container.
-*/
+//! This logger is the logger that's attached to each Holochain application
```

Comment:
why do people like this //! style instead of /*! ... */  This doesn't make sense to me as it means you have to keep fixing the line prefix every time you update the text?

Reply:
In Rust, there are these doc comments. They are actually converted into attributes and serve as a hint for tools like rustdoc. Also they make it possible to check for missing documentation.
So this:
```rust
/// Lorem ipsum...
fn aasdfasdf();
```
actually gets converted into:
```rust
#[doc = "Lorem ipsum..."]
fn aasdfasdf();
```
Also the `!` in attributes means that the enclosing module is affected:
```rust
pub mod something {
    //! This is documentation for the module

    /// This is documentation for the function
    fn blah() {}
}
```
this would be:
```rust
pub mod something {
    #![doc = "This is documentation for the module"]

    #[doc = "This is documentation for the function"]
    fn blah() {}
}
```

Reply:
To make it less painful, the IDE can automate line prefixing. There are also shortcuts for commenting/uncommenting blocks of code.